modules/kafo_configure/lib/puppet/parser/functions/add_progress.rb

```diff
@@ -2,10 +2,15 @@
 
 module Puppet::Parser::Functions
   newfunction(:add_progress) do |args|
-    supported = %w(2.6. 2.7. 3.0. 3.1. 3.2. 3.3. 3.4. 3.5. 3.6. 3.7. 3.8.)
-    if supported.any? { |version| Puppet::PUPPETVERSION.start_with?(version) }
-      # Monkey patch the transaction to put our wrapper around the report object
+    begin
       require 'puppet/transaction'
+      loaded = true
+    rescue LoadError
+      ::Puppet.warning 'Unable to load puppet/transaction for progress bar support, this version may not be supported'
+    end
+
+    if loaded
```

Comment:
I know this works even if require raises an exception since the interpreter already saw `loaded` local variable but could you please add explicit `loaded = false` on before `begin`?